src/test/unit/java/godaddy/registry/jtoolkit2/se/idn/ietf/IetfIdnCommandExtensionTest.java

```diff
@@ -0,0 +1,170 @@
+        Command cmd = new DomainCreateCommand(DOMAIN_NAME, "jtkUT3st");
+        IetfIdnCommandExtension idnExt = new IetfIdnCommandExtension("es", null);
+        cmd.appendExtension(idnExt);
+        String expectedExtension = new StringBuilder("<data xmlns=\"urn:ietf:params:xml:ns:idn-1.0\">")
+                .append("<table>es</table>")
+                .append("</data>")
+                .toString();
+        try {
+            String xml = cmd.toXML();
+            assertEquals(expectedDomainCreateXmlWithExtension(expectedExtension), xml);
+        } catch (SAXException saxe) {
+            fail(saxe.getMessage());
+        }
+    }
+
+    @Test
+    public void failCreateCommandWhenTableNotProvided() throws SAXException {
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("Table must not be null or empty.");
+        Command cmd = new DomainCreateCommand(DOMAIN_NAME, "jtkUT3st");
+        IetfIdnCommandExtension idnExt = new IetfIdnCommandExtension(null, null);
+        cmd.appendExtension(idnExt);
+        cmd.toXML();
+    }
+
+    @Test
+    public void testDomainCheckIdnCommandWithTableAndUname() {
+        Command cmd = new DomainCheckCommand(DOMAIN_NAME);
+        IetfIdnCommandExtension idnExt = new IetfIdnCommandExtension("es", "uname");
+        cmd.appendExtension(idnExt);
+
+        String expectedExtension = new StringBuilder("<data xmlns=\"urn:ietf:params:xml:ns:idn-1.0\">")
+                .append("<table>es</table>")
+                .append("<uname>uname</uname>")
+                .append("</data>")
+                .toString();
+        try {
+            String xml = cmd.toXML();
+            assertEquals(expectedDomainCheckXmlWithExtension(expectedExtension), xml);
+        } catch (SAXException saxe) {
+            fail(saxe.getMessage());
+        }
+    }
+
+
+    @Test
+    public void testDomainCheckIdnCommandWithTableButWithoutUname() {
+        Command cmd = new DomainCheckCommand(DOMAIN_NAME);
+        IetfIdnCommandExtension idnExt = new IetfIdnCommandExtension("es", null);
+        cmd.appendExtension(idnExt);
+        String expectedExtension = new StringBuilder("<data xmlns=\"urn:ietf:params:xml:ns:idn-1.0\">")
+                .append("<table>es</table>")
+                .append("</data>")
+                .toString();
+        try {
+            String xml = cmd.toXML();
+            assertEquals(expectedDomainCheckXmlWithExtension(expectedExtension), xml);
+        } catch (SAXException saxe) {
+            fail(saxe.getMessage());
+        }
+    }
+
+    @Test
+    public void failCheckCommandWhenTableNotProvided() throws SAXException {
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("Table must not be null or empty.");
+        Command cmd = new DomainCheckCommand(DOMAIN_NAME);
+        IetfIdnCommandExtension idnExt = new IetfIdnCommandExtension(null, null);
+        cmd.appendExtension(idnExt);
+        cmd.toXML();
+    }
```

Comment:
Related with my previous comment - we can delete this test case.

Reply:
Deleted